### DISCUSSION TITLE
Possible to Specify Build Path Source?

### DISCUSSION THREAD

**Discussion by fundaxi** [quasarframework/quasar#10990] [CLI - Capacitor / Cordova modes] [answered]
Hey everyone!

I have two projects. One I deploy to PWA and the other to Capacitor for mobile. These two variations share a ton of logic, database connection, api, etc. but the design is vastly different between the two which is why they are in two projects. Therefore, I end up having a ton of duplicated code.

Is it possible to specify the build inputs? I couldn't figure out if any of the build properties in quasar.conf.js would do what I wanted or if it was a webpack thing.

Example:
"quasar build target1" would pull router, pages, components, layouts from specific folders and "quasar build target2" would pull those from a different set of folders.

I appreciate any help regarding this. Thank you!

**Comment by pikil** [comment, marked as answer]
As a suggestion, you can create a folder with sym-linked files/folders. That can save you some time/hair when working with multiple projects.

As an easiest and fastest way - just create another folder inside the project, place all required symlinks in there and create a new npm script mapping in `scripts` (package.json) to that new symlinked folder. E.g. bq1 for `quasar build` # 1 and bq2 for `quasar build` # 2.

Then, ```npm run bq1``` or ```npm run bq2``` should do the trick :)

**Reply by fundaxi** [reply]
Thanks Pikil! I like that suggestion! Could I bother you for a generic example of what that script mapping might look like to get me started?

**Reply by pikil** [reply]
Create another-folder, create links in there, add something similar to package.json (below):

```
"scripts": {
    "bq1": "cd ./another-folder && quasar b",
}
```

Another approach (I should have thought about it beforehand, as it is a bit better, but generally does the same thing and a bit more trickier and limited to quasar build variables):

- Add const config = require(process.env.CONFIG_PATH) to quasar.conf and:

```
"scripts": {
    "bq1": "CONFIG_PATH=./config1 quasar b"
    "bq2": "CONFIG_PATH=./config2 quasar b"
  },
```

**Reply by fundaxi** [reply]
Thanks for both examples! I thought the npm script would be more complicated than that (I'm not too familiar with npm scripts and didn't know I could simply change the directory like that).

I must have missed CONFIG_PATH in the docs! That does seem like the best way. I truly appreciate you guiding me this direction! Thank you again!

**Reply by pikil** [reply]
* CONFIG_PATH is not a part of docs - I made it up :) But the idea is still the same - you just require different files based on the npm script you are calling.

**Reply by fundaxi** [reply]
OH! That's why I missed it! lol.

**Reply by fundaxi** [reply]
**UPDATE**

**Disregard this message! If anyone else was wondering, I had to use cross-env since I'm on Windows (why I stick with Windows is beyond me!)**

**********

Now that I understand the concept, I am having a little trouble with the execution. When I run the script I'm getting

> 'CONFIG_PATH' is not recognized as an internal or external command,
operable program or batch file.

I've looked at the docs referring to adding to process.env (https://quasar.dev/quasar-cli/handling-process-env#adding-to-process-env) but I can't get past this terminal error. Unable to find anything on the forum, stack overflow, google, etc.

Any advice regarding this? Thanks!

**Reply by pikil** [reply]
Cool, glad you make that cracking :)
Just out of a curiosity: did the direct variable assignment work in these scripts

cmd:
`set ENV_CONFIG=config1.json && npm run ...`

or powershell:
`$env:ENV_CONFIG="config1.json" && npm run ...`

**Reply by fundaxi** [reply]
I've tried different variations with bash, powershell, and cmd and if I was error free on the initial script then the dev server would display a blank page like it didn't read the environment variable that I set.

I'm sure one could figure it out, but after I got it to work with cross-env I moved on to my next set of problems. Don't fix it if it isn't broke, right? :)

**Reply by pikil** [reply]
Ah, gotcha :)
